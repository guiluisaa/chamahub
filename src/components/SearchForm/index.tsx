import React, {
  ChangeEvent,
  FC,
  FormEvent,
  useEffect,
  useMemo,
  useState,
} from 'react';

import * as S from './styles';

import Input from '../Input';
import { Link as RouterLink, useLocation } from 'react-router-dom';
import { parse } from 'query-string';

type SearchFormProps = {
  onSubmit: (term: string) => void;
  isLoading?: boolean;
};

const SearchForm: FC<SearchFormProps> = ({ onSubmit, isLoading }) => {
  const [term, setTerm] = useState('');
  const { search } = useLocation();

  const { term: queryTerm } = useMemo(() => parse(search) as { term: string }, [
    search,
  ]);

  useEffect(() => {
    if (queryTerm) setTerm(queryTerm);
  }, []);

  const onChange = (event: ChangeEvent<HTMLInputElement>) =>
    setTerm(event.target.value);

  const submitForm = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    onSubmit(term);
  };

  return (
    <S.Wrapper onSubmit={submitForm}>
      <S.Title>Find an Github user</S.Title>

      <Input
        block
        placeholder="Type the username"
        value={term}
        onChange={onChange}
      />

      <S.Button block isLoading={isLoading} disabled={isLoading}>
        Search
      </S.Button>

      <RouterLink to="/history">
        <S.Link>Terms history</S.Link>
      </RouterLink>
    </S.Wrapper>
  );
};

export default SearchForm;
