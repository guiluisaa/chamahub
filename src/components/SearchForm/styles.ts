import styled from 'styled-components';

import SharedButton from '@/components/Button';
import { Title as SharedTitle } from '@/components/Typograph';

export const Wrapper = styled.form`
  display: block;
  width: 100%;
  max-width: 335px;
  margin: 40px auto;
`;

export const Title = styled(SharedTitle)`
  text-align: center;
  margin-bottom: 40px;
`;

export const Button = styled(SharedButton)`
  margin-top: 20px;
`;

export const Link = styled.a`
  font-family: ${({ theme }) => theme.font.family};
  color: ${({ theme }) => theme.font.color.secondary};
  font-size: 14px;
  line-height: 20px;

  display: block;
  margin: 10px auto 0;
  width: max-content;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;
