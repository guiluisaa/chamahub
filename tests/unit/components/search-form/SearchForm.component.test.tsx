import React from 'react';
import { shallow } from 'enzyme';

import SearchForm from '@/components/search-form/SearchForm.component';

jest.mock('@/io/user/useUserRepos.hook', () => () => ({
  repos: [],
  getUserRepos: jest.fn(),
  isLoading: false,
}));

jest.mock('react-router-dom', () => ({
  useLocation: () => ({ sarch: '' }),
}));

describe('<SearchForm />', () => {
  it('should match snapchat', () => {
    const onSubmit = jest.fn();

    const wrapper = shallow(<SearchForm onSubmit={onSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });
});
