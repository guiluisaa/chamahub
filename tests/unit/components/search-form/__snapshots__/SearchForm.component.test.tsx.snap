// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> should match snapchat 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <Styled(styled.h2)>
    Find an Github user
  </Styled(styled.h2)>
  <styled.input
    block={true}
    onChange={[Function]}
    placeholder="Type the username"
    value=""
  />
  <Styled(styled.button)
    block={true}
  >
    Search
  </Styled(styled.button)>
  <Component
    to="/history"
  >
    <styled.a>
      Terms history
    </styled.a>
  </Component>
</styled.form>
`;
